_posts/2020-07-27-cpp-build-simplification.md

```diff
@@ -0,0 +1,271 @@
+requirements spill over into their project's requirements. Indeed, in the
+earlier stages of the Arrow project development, dependency management issues
+did cause problems for early adopters.
+
+We want developers to trust that they can use and depend on our libraries, and
+that doing so doesn't add a burden for their own project maintenance or for
+their users. Over the last year, we have undertaken a number of significant
+projects to accommodate the different ways that people want to depend on Arrow
+C++. We've aimed to make the build process simple by default, without requiring
+special environment setup, yet also highly configurable for those who need to
+specialize. And we've worked to make builds faster and more compact, even as we
+continue to add new functionality.
+
+This post covers many of the efforts we've made, both in the C++ libraries and
+in the Arrow Python and R packages that depend on them. Compared to a year ago,
+the build experience is much more reliable on a wider range of platforms,
+requires fewer dependencies, and yields smaller package sizes.
+
+## Minimal default build options
+
+One rough edge for people using Arrow as a dependency was that many optional
+project components were enabled in the build by default, thus requiring any
+extra dependencies of those optional components. Rather than expecting users to
+disable optional components one by one, we have made the default for all
+optional components to be `OFF` so that the default configuration is a
+dependency-free minimal core build.
+
+The only third-party library enabled by default is jemalloc, the project's
+recommended memory allocator. Given that Arrow applications often process large
+volumes of data, we have found additionally that using memory allocators
+provided by projects like jemalloc and mimalloc yield significantly better
+performance over the default system allocators. Even so, this can also be
+disabled if desired.
+
+To demonstrate a minimal build, we have provided a
+[Dockerfile](https://github.com/apache/arrow/blob/master/cpp/examples/minimal_build/Dockerfile)
+which can be used to build the project requiring only CMake and a C++ compiler
+with zero dependencies. Additionally, we have included an
+[example](https://github.com/apache/arrow/tree/master/cpp/examples/minimal_build)
+of including Arrow as an external project dependency in another CMake project.
+
+## Flexible dependency configuration in CMake
+
+As part of improving our CMake-based build system, we have made the
+configuration of build dependencies both flexible and consistent for different
+users' needs. In some cases, developers want Arrow to build against dependencies
+provided by an external package manager, such as apt in Debian-based Linux
+distributions. In other cases, developers may want to avoid any quirks of system
+libraries and build all dependencies together with the Arrow build.
+
+For each package, the `$Library_SOURCE` CMake option can be set to one of three
```

Comment:
Some readers may think whole `$Library_SOURCE` is variable.
How about using `${Library}` instead of `$Library`?

```suggestion
For each package, the `${Library}_SOURCE` CMake option can be set to one of three
```